Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorApplyMotionSetComponent.cpp

```diff
@@ -0,0 +1,324 @@
+/*
+* All or portions of this file Copyright (c) Amazon.com, Inc. or its affiliates or
+* its licensors.
+*
+* For complete copyright and license terms please see the LICENSE at the root of this
+* distribution (the "License"). All use of this software is governed by the License,
+* or, if provided, by the license below or the license accompanying this file. Do not
+* remove or modify any license notices. This file is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*
+*/
+
+
+#if defined(CARBONATED)
+#include <AzCore/Asset/AssetManager.h>
+#include <AzCore/Script/ScriptProperty.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzToolsFramework/API/ToolsApplicationAPI.h>
+#include <EMotionFX/Source/Parameter/BoolParameter.h>
+#include <EMotionFX/Source/Parameter/FloatParameter.h>
+#include <EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.h>
+#include <EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/MainWindow.h>
+#include <Integration/ActorComponentBus.h>
+#include <Integration/Editor/Components/EditorApplyMotionSetComponent.h>
+#include <EMotionFX/Source/MotionSet.h>
+
+#include <QApplication>
+
+
+namespace EMotionFX
+{
+    namespace Integration
+    {
+        //////////////////////////////////////////////////////////////////////////
+        void EditorApplyMotionSetComponent::Reflect(AZ::ReflectContext* context)
+        {
+            auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+            if (serializeContext)
+            {
+                serializeContext->Class<EditorApplyMotionSetComponent, AzToolsFramework::Components::EditorComponentBase>()
+#if defined(CARBONATED)

```

Comment:
Looks like inner #ifdef/#endif for CARBONATED are not necessary
here and below
due to we use #if defined(CARBONATED) for whole file in the line 14
?